hs_core/hydroshare/resource.py

```diff
@@ -1001,54 +924,66 @@ def publish_resource(user, pk):
                               "metadata or content files or this resource type is not allowed "
                               "for publication.")
 
-    # append pending to the doi field to indicate DOI is not activated yet. Upon successful
-    # activation, "pending" will be removed from DOI field
-    resource.doi = get_resource_doi(pk, 'pending')
+    if istorage.exists(res_coll):
+        bag_modified = istorage.getAVU(res_coll, 'bag_modified')
+        if bag_modified.lower() == "true":
+            hs_bagit.create_bag(resource)
+    else:
+        raise ValidationError("Resource {} does not exist in iRODS".format(resource.short_id))
+
+    # create minid for the bag, using the checksum of the zip
+    tmpdir = os.path.join(settings.TEMP_FILE_DIR, uuid4().hex)
+    tmpfile = os.path.join(tmpdir, 'bag.zip')
+    os.makedirs(tmpdir)
+
+    bag_full_name = 'bags/{res_id}.zip'.format(res_id=resource_id)
+    irods_dest_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    srcfile = os.path.join(irods_dest_prefix, bag_full_name)
+    istorage.getFile(srcfile, tmpfile)
+    checksum = mca.compute_checksum(tmpfile)
+    resource_url = '{0}/resource/{1}'.format(utils.current_site_url(), resource.short_id)
+    download_bag_url = '{0}/django_irods/download/bags/{1}.zip'.format(utils.current_site_url(), resource.short_id)
+    locations = [resource_url, download_bag_url]
+    config= mca.parse_config('config/minid-config.cfg')
+    minid = mca.register_entity(config['minid_server'],
+                                checksum,
+                                config['email'],
+                                config['code'],
+                                locations, 'MINID for ' + resource.title, True)
+    resource.minid = minid
     resource.save()
```

Comment:
Make sure to clean up the tmpdir after it serves its purpose every time after it is created to avoid left-over temp files in the system.